src/pages/pool/_id.vue

```diff
@@ -88,6 +90,24 @@ const isLoadingSnapshots = computed(() => poolSnapshotsQuery.isLoading.value);
 const snapshots = computed(() => poolSnapshotsQuery.data.value);
 //#endregion
 
+//#region pool snapshot query
+const poolMetadataQuery = useMetadataQuery(poolId);
+const isLoadingMetadata = computed(() => poolMetadataQuery.isLoading.value);
+
+const poolMetadataCID = computed(() => {
+  if (!poolMetadataQuery.data.value) return null;
+  return poolMetadataQuery.data.value.metadataCID;
+});
+
+async function getPoolMetadata() {
+  if (!poolMetadataCID || !poolMetadataCID.value) return '';
+
+  return await ipfsService.get(poolMetadataCID.value).then(res => {
+    return res;
+  });
+}
```

Comment:
Would it make sense for this to also be done in the MetadataQuery to expose a nicer interface?

Reply:
yes! wasn't so confident in doing this on the first try, but today is much clearer what has to be done to achieve this. Implementing asap